packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -0,0 +1,158 @@
+  /// which creates the platform view and manages its lifecycle.
+  int getNextPlatformViewId() => _nextPlatformViewId++;
+}
+
+/// Callback signature for when a platform view was created.
+///
+/// `id` is the platform view's unique identifier.
+typedef void OnPlatformViewCreated(int id);
+
+/// Provides access to the platform views service.
+///
+/// This service allows creating and controlling Android views.
+///
+/// See also: [PlatformView].
+class PlatformViewsService {
+  PlatformViewsService._();
+
+  /// Creates a controller for a new Android view.
+  ///
+  /// `id` is an unused unique identifier generated with [platformViewsRegistry].
+  ///
+  /// `viewType` is the identifier of the Android view type to be created, a
+  /// factory for this view type must have been registered on the platform side.
+  /// Platform view factories are typically registered by plugin code.
+  /// Plugins can register a platform view factory with
+  /// [PlatformViewRegistry#registerViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html#registerViewFactory-java.lang.String-io.flutter.plugin.platform.PlatformViewFactory-).
+  ///
+  /// The Android view will only be created after [AndroidViewController.setSize] is called for the
+  /// first time.
+  static AndroidViewController initAndroidView({
+    @required int id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) {
+    return new AndroidViewController._(
+        id: id,
+        viewType: viewType,
+        onPlatformViewCreated: onPlatformViewCreated
+    );
+  }
+}
+
+enum _AndroidViewState {
+  waitingForSize,
+  creating,
+  created,
+  createFailed,
+  disposed,
+}
+
+/// Controls an Android view.
+///
+/// Typically created with [PlatformViewsService.initAndroidView].
+class AndroidViewController {
+  AndroidViewController._({
+    @required this.id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) : assert(id != null),
+       assert(viewType != null),
+       _viewType = viewType,
+        _onPlatformViewCreated = onPlatformViewCreated,
+       _state = _AndroidViewState.waitingForSize;
+
+  /// The unique identifier of the Android view controlled by this controller.
+  final int id;
+
+  final String _viewType;
+
+  final OnPlatformViewCreated _onPlatformViewCreated;
+
+  /// The texture entry id into which the Android view is rendered.
+  int _textureId;
+
+  /// Returns the texture entry id that the Android view is rendering into.
+  ///
+  /// Returns null if the Android view has not been successfully created, or if it has been
+  /// disposed.
+  int get textureId => _textureId;
+
+  _AndroidViewState _state;
+
+  /// Disposes the Android view.
+  ///
+  /// The [AndroidViewController] object is unusable after calling this.
+  /// The identifier of the platform view cannot be reused after the view is
+  /// disposed.
+  Future<void> dispose() async {
```

Comment:
Should this return a boolean in case users want to know whether disposing succeeded? (since it'll be hard to change this API after)

Reply:
The future will complete with a PlatformException in that case, isn't that enough?

Reply:
Ah, makes sense.

Reply:
If that is part of the expected API, you should add a doc comment like:

/// Throws [PlatformException] if ...

Reply:
I don't really see it happens, unless something really blows up on the platform side.
I can see that it makes sense to add such comments when there's an obvious failure case.
Everytime we invokeMethod on a method channel we can get a PlatformException... (I guess theoretically you can always get some exception thrown if something really bad happens) randomly sampling a few call sites to invokeMethod in our codebase non of them mentions in dartdoc that an exception might be thrown.

Reply:
If that exception doesn't indicate any useful information besides "something broke" then I wouldn't include it